protocols/readme.md

```diff
@@ -0,0 +1,29 @@
+## Off-chain protocols
+
+This package defines the off-chain protocols for managing channels. These protocols are abstract rules which clients should follow in order to reach a desired goal.
+
+To implement the protocols, we make use of an abstraction we call the `Objective`. Objectives are named for the goal which the off-chain protocol aspires to achieve.
+
+The following table shows some example objectives and their implemetation status:
+
+| Objective type      | Implemented |
+| ------------------- | ----------- |
+| `direct-funding`    | x           |
+| `direct-defunding`  |             |
+| `virtual-funding`   |             |
+| `virtual-defunding` |             |
+| `challenge`         |             |
+
+The set of objectives comprises the functional core of a go-nitro client. They expose only _pure_ functions -- but otherwise take on as much responsibility as possible, leaving only a small amount of responsibility to an imperative shell.
+
+Each `Objective` type is implemented by a go class implementing the `Objective` interface. Each instance of an `Objective` type will hold its own data.
+
+The imperative shell will typically be responsible for:
+
+- spawning new objective instances of one type or another
+- reading and persisting objectives to and from a store.
+- "updating" objectives by passing events to their `Update(event)` function (an updated copy is returned)
+- "cranking" objectives by calling `Crank()` (side effects are returned)
+- executing side effects
+
+There is an architectural decision record contained in a hidden folder in this directory. [Link](./.adr/readme.md)
```

Comment:
```suggestion
This folder contains a hidden directory with architectural decision records. [Link](./adr)
```

Plus a rename of `.adr/readme.md` to something like `0000-flowchart-vs-statechart.md`.